### PR TITLE
Add `_PaddingLayout` support to layout engine

### PR DESCRIPTION
**Pull request by carson-katri**
This adds support for the `padding` modifier to the custom layout engine.

### CODE REVIEW

File: Sources/TokamakStaticHTML/Modifiers/LayoutModifiers.swift [hunk 1]
```diff
@@ -144,6 +144,14 @@ extension _PaddingLayout: DOMViewModifier {
   }
 }
 
+@_spi(TokamakStaticHTML) extension _PaddingLayout: HTMLConvertible {
```
Comment (MaxDesiatov) [inline Sources/TokamakStaticHTML/Modifiers/LayoutModifiers.swift:147]:
```suggestion
@_spi(TokamakStaticHTML) 
extension _PaddingLayout: HTMLConvertible {
```

File: Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      top: edges.contains(.top) ? insets.top : 0,
+      leading: edges.contains(.leading) ? insets.leading : 0,
+      bottom: edges.contains(.bottom) ? insets.bottom : 0,
+      trailing: edges.contains(.trailing) ? insets.trailing : 0
+    )
+  }
+}
+
+/// A `LayoutComputer` that fits to its children then adds padding.
+final class PaddingLayoutComputer: LayoutComputer {
```
Comment (MaxDesiatov) [inline Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift:32]: I think I had a similar comment before, but got no clarity on this yet: are types conforming to `LayoutComputer` supposed to be classes or structs? If structs are fine, let's avoid classes without an explicit need for them, which helps us avoid slower allocations on the heap.

File: Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  }
+}
+
+public extension _PaddingLayout {
+  static func _makeView(_ inputs: ViewInputs<Self>) -> ViewOutputs {
+    .init(inputs: inputs, layoutComputer: { PaddingLayoutComputer(
+      proposedSize: $0,
+      edges: inputs.content.edges,
+      insets: inputs.content.insets
+    ) })
```
Comment (MaxDesiatov) [inline Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift:77]: nit: is this formatting applied by SwiftFormat?

File: Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift (hunk 2 above)
Comment (j-f1) [reply Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift:32]: …and if they should be classes, then the `LayoutComputer` protocol should inherit from `AnyObject` to ensure that.

File: Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift (hunk 3 above)
Comment (carson-katri) [reply Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift:77]: Must be, I don't remember doing this.

File: Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift (hunk 2 above)
Comment (carson-katri) [reply Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift:32]: They should be structs now, I has changed that before merging the layout branch, but had made this PR before that change was made.

**Review by MaxDesiatov — APPROVED** [review]
Thanks 🙏

**Pull request merged**